specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/vaults.json

```diff
@@ -0,0 +1,1803 @@
+      "description": "Vault extended information.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/VaultExtendedInfo",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "VaultList": {
+      "description": "The response model for a list of Vaults.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Vault"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "VaultProperties": {
+      "description": "Properties of the vault.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning State.",
+          "type": "string",
+          "readOnly": true
+        },
+        "upgradeDetails": {
+          "$ref": "#/definitions/UpgradeDetails"
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnectionVaultProperties"
+          },
+          "description": "List of private endpoint connection.",
+          "readOnly": true
+        },
+        "privateEndpointStateForBackup": {
+          "description": "Private endpoint state for backup.",
+          "enum": [
+            "None",
+            "Enabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "VaultPrivateEndpointState",
+            "modelAsString": true
+          }
+        },
+        "privateEndpointStateForSiteRecovery": {
+          "description": "Private endpoint state for site recovery.",
+          "enum": [
+            "None",
+            "Enabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "VaultPrivateEndpointState",
+            "modelAsString": true
+          }
+        },
+        "encryption": {
+          "description": "Customer Managed Key details of the resource.",
+          "type": "object",
+          "properties": {
+            "keyVaultProperties": {
+              "$ref": "#/definitions/CmkKeyVaultProperties"
+            },
+            "kekIdentity": {
+              "$ref": "#/definitions/CmkKekIdentity"
+            },
+            "infrastructureEncryption": {
+              "description": "Enabling/Disabling the Double Encryption state",
+              "type": "string",
+              "enum": [
+                "Enabled",
+                "Disabled"
+              ],
+              "x-ms-enum": {
+                "name": "InfrastructureEncryptionState",
+                "modelAsString": true
+              }
+            }
+          }
+        },
+        "moveDetails": {
```

Comment:
Why is this part of the resource?  The move contract is handled at the ARM level (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#move-resource)

Reply:
Thanks

Reply:
👍

Reply:
We were flagged in the s360 that the Get resource sends the move details but it was not present in the swagger

Reply:
All the fields are marked read only

Reply:
Also, what is the correct guidance here for this, because if the resource is being moved, and a user does a get resource, we send these details as added in the swagger

Reply:
Is move information needed by other service? 
As Chris mentioned, resource move is an ARM level contract, so depending on your implementation, you can prevent move details from being returned in a get response (e.g. DTO)

Reply:
This is because in the Azure Portal when the resource is moved we display all these information as a part of GET Vault call in case of any issues

Reply:
If the resource goes through a series of move executions, I assume the answer is yes but want to make sure, you always returns the latest resource move information correct? If yes, I'd suggest adding it to the description so it is crystal clear for users.

Reply:
It is the latest one, will be updating the description for clarity

Reply:
Updated